enterprise/cmd/frontend/worker/auth/perms_syncer_scheduler_test.go

```diff
@@ -0,0 +1,198 @@
+	logger := logtest.Scoped(t)
+	db := database.NewDB(logger, dbtest.NewDB(logger, t))
+
+	store := database.PermissionSyncJobsWith(logger, db)
+	usersStore := database.UsersWith(logger, db)
+	reposStore := database.ReposWith(logger, db)
+	permsStore := edb.Perms(logger, db, clock)
+
+	// Creating site-admin.
+	_, err := usersStore.Create(ctx, database.NewUser{Username: "admin"})
+	require.NoError(t, err)
+
+	// Creating non-private repo.
+	nonPrivateRepo := types.Repo{Name: "test-public-repo"}
+	err = reposStore.Create(ctx, &nonPrivateRepo)
+	require.NoError(t, err)
+
+	// We should have no jobs scheduled
+	runJobsTest(t, db, store, []testJob{})
+
+	// Creating a user.
+	user1, err := usersStore.Create(ctx, database.NewUser{Username: "test-user-1"})
+	require.NoError(t, err)
+
+	// Creating a repo.
+	repo1 := types.Repo{Name: "test-repo-1", Private: true}
+	err = reposStore.Create(ctx, &repo1)
+	require.NoError(t, err)
+
+	// We should have 2 jobs scheduled.
+	wantJobs := []testJob{
+		{
+			UserID:       int(user1.ID),
+			RepositoryID: 0,
+			Reason:       database.ReasonUserNoPermissions,
+			Priority:     database.MediumPriorityPermissionSync,
+		},
+		{
+			UserID:       0,
+			RepositoryID: int(repo1.ID),
+			Reason:       database.ReasonRepoNoPermissions,
+			Priority:     database.MediumPriorityPermissionSync,
+		},
+	}
+	runJobsTest(t, db, store, wantJobs)
+
+	// Touch perms for the user and repo.
+	permsStore.TouchUserPermissions(ctx, user1.ID)
+	permsStore.TouchRepoPermissions(ctx, int32(repo1.ID))
+
+	// We should have same 2 jobs because jobs with higher priority already exists.
+	runJobsTest(t, db, store, wantJobs)
+
+	// Creating a user.
+	user2, err := usersStore.Create(ctx, database.NewUser{Username: "test-user-2"})
+	require.NoError(t, err)
+
+	// Creating a repo.
+	repo2 := types.Repo{Name: "test-repo-2", Private: true}
+	err = reposStore.Create(ctx, &repo2)
+	require.NoError(t, err)
+
+	// Touch perms for the user and repo.
+	permsStore.TouchUserPermissions(ctx, user2.ID)
+	permsStore.TouchRepoPermissions(ctx, int32(repo2.ID))
+
+	// We should have same 4 jobs scheduled including new jobs for user2 and repo2.
+	wantJobs = []testJob{
+		{
+			UserID:       int(user1.ID),
+			RepositoryID: 0,
+			Reason:       database.ReasonUserNoPermissions,
+			Priority:     database.MediumPriorityPermissionSync,
+		},
+		{
+			UserID:       0,
+			RepositoryID: int(repo1.ID),
+			Reason:       database.ReasonRepoNoPermissions,
+			Priority:     database.MediumPriorityPermissionSync,
+		},
+		{
+			UserID:       int(user2.ID),
+			RepositoryID: 0,
+			Reason:       database.ReasonUserOutdatedPermissions,
+			Priority:     database.LowPriorityPermissionSync,
+		},
+		{
+			UserID:       0,
+			RepositoryID: int(repo2.ID),
+			Reason:       database.ReasonRepoOutdatedPermissions,
+			Priority:     database.LowPriorityPermissionSync,
+		},
+	}
+	runJobsTest(t, db, store, wantJobs)
+
+	// Set user1 and repo2 schedule jobs to completed.
```

Comment:
According to the actual code below: 
```suggestion
	// Set user1 and repo1 schedule jobs to completed.
```

Reply:
done